netlogo-core/src/main/api/Workspace.scala

```diff
@@ -85,4 +85,12 @@ with ViewSettings with Controllable {
   def renderer: RendererInterface
   def compilerTestingMode: Boolean
   def warningMessage(message: String): Boolean
+
+  @throws(classOf[CompilerException])
+  def evaluateCommands(owner: JobOwner, source: String): Unit
+  @throws(classOf[CompilerException])
+  def evaluateCommands(owner: JobOwner, source: String, waitForCompletion: Boolean): Unit
+
+  @throws(classOf[CompilerException])
+  def evaluateReporter(owner: JobOwner, source: String): AnyRef
```

Comment:
The methods `command` and `report` might be more useful for normal use. They don't require a `JobOwner`, which makes them much easier to use (though creating a `SimpleJobOwner` isn't hard, it does complicate the process unnecessarily).